ERCS/erc-7651.md

```diff
@@ -0,0 +1,349 @@
+
+### ERC-165 Interface
+
+We include the [ERC-165](./eip-165.md) interface in specification both to adhere to [ERC-721](./eip-721.md) design philosophy, and as a means of exposing interfaces at the contract level. We see this as a valuable, accepted standard to adhere to such that integrating applications may identify underlying specification.
+
+Note that [ERC-7651](./eip-7651.md) contracts should not make any claim through `supportsInterface` to support [ERC-721](./eip-721.md) or [ERC-20](./eip-20.md) standards as, despite strong backwards compatibility efforts, these contracts cannot fully adhere to existing specifications.
+
+### Metadata
+
+In-line with [ERC-721](./eip-721.md), we've decided to isolate replicated metadata functionality through a separate interface. This interface includes traditional naming and token URI logic, though also introduces patterns surrounding token banking visibility, as outlined above in both the NFT Banking and Transfer Logic sections.
+
+## Backwards Compatibility
+
+The fractional non-fungible token standard aims to be nearly backwards compatible with existing [ERC-721](./eip-721.md) and [ERC-20](./eip-20.md) standards, though makes no claim to fully adhere to either and has as such been proposed through a distinct interface.
+
+### Events
+
+Events in [ERC-721](./eip-721.md) and [ERC-20](./eip-20.md) specifications share conflicting signatures on approval and transfer, meaning an adherent hybrid of the two cannot be achieved.
+
+This is one of the few areas where backwards compatibility has been intentionally broken, resulting in a new series of events with either a `Fractional` or `NonFungible` prefix. We believe that a decisive move to a non-conflicting, descriptive solution is ideal here, though will require external lift for indexing software.
+
+### balanceOf
+
+The `balanceOf` function as defined in both [ERC-20](./eip-20.md) and [ERC-721](./eip-721.md) standards varies, in practice, to represent either fractional or whole token ownership respectively. Given fractional non-fungible tokens should adhere to an underlying fractional representation, it follows that this function should return a balance in that representation. This does, however, imply that fractional NFT contracts cannot fully adhere to the `balanceOf` specification provided by ERC-721.
+
+### Success Return Values
+
+The `transfer` and `approve` functions both return a boolean value indicating success or failure. This is non-standard for the [ERC-721](./eip-721.md) specification, though is standard for [ERC-20](./eip-20.md). Fractional non-fungible tokens adhere to a returned boolean value to meet minimum expectations for the ERC-20 standard, acknowledging that this deviates from a state of ideal backwards compatibility.
+
+## Security Considerations
+
+Needs discussion.
```

Comment:
I think given some interface and events are identical, we shall consider a security issue when an ERC-721 or ERC-20 client (dApp/wallet/DEX) will unintentionally misinterpret a contract of this ERC as ERC-721 or ERC-20, such as Uniswap, what will happen. Given the potential financial implication, please include a least a placeholder for such discussion before checking in the Draft

Reply:
I agree. The problem is particularly relevant when discussing the fungible component because ERC-20 does not extend ERC165, so exchanges may assume that an ERC7651 is actually an ERC20. I see fewer issues with ERC721 because exchanges like OpenSea check if the token supports the IERC721 interface.

Reply:
Totally agree, we've given a lot of consideration to this, though I know there are still a few edgecases. I can kick off discussion here shortly by raising a few. In the meantime, added a more direct placeholder here that calls out the need for discussion surrounding interface misinterpretation.